clients/python/src/osparc/_utils.py

```diff
@@ -46,38 +48,40 @@ def __init__(
         self._client: httpx.Client = httpx.Client(
             auth=auth, base_url=base_url, follow_redirects=True
         )
+        self._page: Optional[Page] = None
+        self._page_item_counter = 0
 
     def __del__(self):
         self._client.close()
 
+    def __next__(self):
+        if self._page is None:
+            self._page = self._first_page_callback()
+        if self._page_item_counter > (len(self._page.items) - 1):
+            next_page_url = self._page.links.next
+            if next_page_url is None:
+                self._page = None
+                self._page_item_counter = 0
+                raise StopIteration
+            response = self._client.get(next_page_url)
```

Comment:
Q: why not using async clients as well?

Reply:
Hmm, I could do that yes. I think especially for the file upload it makes a lot of sense because chunks can be uploaded in parallel, but here I would wait until someone actually needs it to be async.